ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,305 @@
+
+from rosidl_runtime_py import get_interface_path
+
+
+_TEMPLATES_DIR = 'templates'
+IGNORED_KEYS = ['__slots__', '__doc__', '_fields_and_field_types', 'SLOT_TYPES', '__init__',
+                '__repr__', '__eq__', '__hash__', 'get_fields_and_field_types', '__module__']
+
+
+def generate_raw_text(raw_text):
+    """
+    Generate the text HTML for the message fields.
+
+    Lines that starts with '#' (comments) will be shown in blue, otherwise black.
+
+    Parameters
+    ----------
+    raw_text: str
+        content of the interface
+    Returns
+    -------
+    raw_test_str: str
+        string with the generated HTML
+
+    """
+    raw_test_str = ''
+    for line in raw_text.split(os.linesep):
+        line = line.replace(' ', '&nbsp;')
+        parts = line.split('#')
+        if len(parts) > 1:
+            raw_test_str = raw_test_str + parts[0] \
+                + '<font color="blue">#%s</font><br/>' % ('#'.join(parts[1:]))
+        else:
+            raw_test_str = raw_test_str + '%s<br/>' % parts[0]
+    return raw_test_str
+
+
+def resource_name(resource):
+    """
+    Return the resource name.
+
+    Parameters
+    ----------
+    resource: str
+        resource name of the interface
+
+    Returns
+    -------
+    value: tuple
+        a tuple with the 3 part of the resource name (for example: std_msgs/msg/Bool ->
+        ('std_msgs', 'msg', 'Bool'))
+
+    """
+    if '/' not in resource:
+        return '', '', resource
+    values = resource.split('/')
+    if len(values) != 3:
+        raise ValueError('Resource name "{}" is malformed'.format(resource))
+    return tuple(values)
+
+
+def get_templates_dir():
+    """
+    Return template directory.
+
+    Returns
+    -------
+    template directory:  str
+        return the directory of the template directory
+
+    """
+    return os.path.join(os.path.dirname(__file__), _TEMPLATES_DIR)
+
+
+def copy_css_style(folder_name):
+    """
+    Copy the css style files in the folder_name.
+
+    Parameters
+    ----------
+    folder_name: str
+        name of the folder where the style css files will be copied
+
+    """
+    style_css, _ = load_template('styles.css')
+    with open(os.path.join(folder_name, 'styles.css'), 'w') as f:
+        f.write(style_css)
+
+    msg_style_css, _ = load_template('msg-styles.css')
+    with open(os.path.join(folder_name, 'msg-styles.css'), 'w') as f:
+        f.write(msg_style_css)
+
+
+def load_template(filename):
+    """
+    Look up file within rosdoc ROS package.
+
+    return its content, may sys.exit on error.
+    Contents are cached with filename as key.
+
+    Parameters
+    ----------
+    filename: str
+        name of the template to be loaded
+
+    Returns
+    -------
+    content: str
+        cached file contents
+    filename: str
+        filename of the template concat with the template directory
+
+    """
+    filename = os.path.join(get_templates_dir(), filename)
```

Comment:
Changing the meaning of the parameter / variable seems weird. Using a different name would be more readable.